File: examples-sdk/csharp/401-agent-calculator/Main.cs
// 401-agent-calculator — C# SDK equivalent
// Compile: vil compile --from csharp --input 401-agent-calculator/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("calculator-agent", 3120);
var calc_agent = new ServiceProcess("calc-agent");
calc_agent.Endpoint("POST", "/calc", "calc_handler");
server.Service(calc_agent);
server.Compile();

File: examples-sdk/swift/401-agent-calculator/main.swift
// 401-agent-calculator — Swift SDK equivalent
// Compile: vil compile --from swift --input 401-agent-calculator/main.swift --release

let server = VilServer(name: "calculator-agent", port: 3120)
let calc_agent = ServiceProcess(name: "calc-agent")
calc_agent.endpoint(method: "POST", path: "/calc", handler: "calc_handler")
server.service(calc_agent)
server.compile()
